### ISSUE TITLE
Expose converged per-token coordinates + assignments from sae_manifold_fit (+ warm-start init) for amortized-encoder distillation

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#357]
## Context
Building an **amortized Manifold-SAE**: a feedforward encoder predicts, per token, the sparse atom assignments and the on-manifold coordinates `t`, so inference is a single forward pass (needed for LLM scale: F~100K atoms, billions of tokens). The gam joint Arrow-Schur solve (`sae_manifold_fit`) is the natural *teacher* — it produces the optimal coordinates/assignments — and the encoder amortizes it (Predictive-Sparse-Decomposition / iterative-amortized-inference style). For that to work, the fit needs to expose its converged latents and accept a warm start.

## Requests
1. **Expose the converged per-token latents from `sae_manifold_fit`.** After the joint solve, return (or attach to the result) the per-token, per-atom **on-manifold coordinates `t*` (N, K, d)** and the **assignments / gate `a*` (N, K)** that the solver converged to — not just the decoder blocks `B_k`. These are the supervision targets for the amortized encoder.
2. **Warm-start + bounded refinement.** Let `sae_manifold_fit` accept an initial guess for coordinates `t_init` and assignments `a_init` (encoder predictions) and run a small bounded number of refinement steps (`n_iter=k`), so training can do "encoder predicts → solver refines a few steps → distill the gap." (`Z=` exists in the signature — please document whether it is this warm-start, and for which quantity.)
3. **Standalone per-atom projection** `project(x, atom_k) -> t` (ambient point → its on-manifold coordinate for atom k). This is the minimal teacher signal for the coordinate head and is useful without running the full joint solve.

(1)+(2) together enable the amortized-inference training loop; (3) is a lighter-weight fallback teacher. Blocked in practice by #356 (the fit currently panics), but these are the interface additions needed once that lands.

**Comment by SauersML** [comment]
## Code-grounded refinement (after reading the solver)

The warm-startable, bounded-step core for this **already exists** — it just isn't reachable from Python. Concretely, in `crates/gam-pyffi/src/lib.rs`:

- `sae_manifold_fit_inner` (~L9417) already accepts the full warm-start state as inputs: `decoder_coefficients`, `initial_logits`, `initial_coords`, and honors `max_iter` (so `max_iter=1..3` = bounded refinement). It is a plain `fn`, **not** a `#[pyfunction]`.
- The only Python-reachable fitters — `sae_manifold_fit_minimal` (~L11072), `_ibp`, `_auto` — **cold-seed every call**: PCA seed via `sae_pca_seed_initial_coords` + least-squares decoder via `sae_decoder_lsq_init`, discarding any prior state.
- The inner result dict sets: `atoms`, `logits`, `assignments_z`, `atom_active_mask`, `fitted`, `log_ard`, `log_lambda_smooth`, `reml_score`, `active_dim`, `basis_centers`, `basis_kind`, `assignment_prior`. **Missing: the converged per-token coordinates** (the `initial_coords` input has no `coords`/`t` counterpart in the output), so positions can't be threaded back even today.

### Concrete ask
Expose a stateful continue entry — e.g. `sae_manifold_fit_continue(z, state, max_iter, ...)`, or `warm_state=` + `return_state=` flags on the existing minimal fitter — that:

1. accepts prior state (`atoms`/decoder blocks, per-token `coords`, assignment `logits`, `log_ard`, `log_lambda_smooth`) and **skips the PCA + lsq cold seed when it's provided**;
2. runs a small `max_iter` (already supported);
3. returns the **full** state, including per-token coordinates, so it can be fed straight back on the next minibatch.

### Why
This turns the joint Arrow-Schur solve from a batch cold re-solve into a **per-minibatch refinement that folds into a training loop** — the M-step of an amortized-EM SAE where a feedforward encoder is distilled to the warm-continued solve, instead of running a full solve as a separate phase. The Rust core already does the optimization; this is plumbing at the FFI boundary (expose the inner driver + return coords + honor a warm seed).

Sequencing: blocked behind #356 (the inner path currently panics with an `[N*D, M] -> [N, M]` broadcast); complements #358 (the same continue entry needs GPU + minibatch to be useful at scale).

**Comment by SauersML** [comment]
Closing — feature landed in 34b37eff7 and now has a real regression test (bda4c877a). All three requests implemented and verified:

**1. Converged per-token latents (t*, a*).** `ManifoldSAE.converged_latents(X=None)` returns `{coords: [(N, d_k)…], assignments: (N, K), logits: (N, K), fitted: (N, p)}` — the per-atom on-manifold coordinates `t*` and the assignments/gate `a*` the joint solver converged to, not just the decoder blocks. With `X=None` (or training `X`) it returns the stored training-fit latents; for fresh `X` it runs the frozen-decoder OOS solve. `sae_manifold_predict_oos` now returns the full payload (a*, per-atom t*, logits, fitted) instead of only the reconstruction (gamfit/_sae_manifold.py:187-215; crates/gam-pyffi/src/lib.rs OOS payload).

**2. Warm-start + bounded refinement.** `sae_manifold_fit` accepts `a_init` (assignment logits, (N, K)) and `t_init` (on-manifold coords, (K, N, D_max)), eagerly shape-validated, and threads them into `sae_manifold_fit_minimal` / `sae_manifold_predict_oos` as the Newton start (`start_coords` / `warm_logits`), falling back to the PCA seed when absent. The bounded refinement is `n_iter` (encoder-predicts → solver-refines-k-steps → distill). `Z` is clarified in the docstring and code as the *data alias*, **not** a warm start. (gamfit/_sae_manifold.py:625-651; crates/gam-pyffi/src/lib.rs:11262-11300, 11316-11334.)

**3. Standalone per-atom projection.** `fit.project(x, atom_k) -> t` maps an ambient point to its on-manifold coordinate for atom `k` via the frozen-decoder OOS solve, returning the `(N, d_k)` block; `IndexError` out of range (gamfit/_sae_manifold.py:217-231).

**Regression coverage:** new `tests/test_sae_manifold_converged_latents_issue_357.py` pins all three on the public surface — converged_latents exposes t*/a* equal to the solver's own attributes on training X; project returns the per-atom coordinate and rejects out-of-range k; a_init/t_init warm-start a bounded refinement (seeded from a teacher fit's converged latents) and are shape-validated (N,K)/(K,N,D_max) distinct from Z.

The original block on #356 is resolved (closed). Principled, no shim, no back-compat layer (the supervised caller was migrated atomically to the OOS payload). Meets the bar.

**Comment by SauersML** [comment]
Reopening: this was closed as fixed in 34b37eff7 with a regression test (bda4c877a), but the regression test **panics at runtime** — the feature it pins is unreachable for any fit whose structure search grows the atom count.

**Repro** (current `main`): `pytest tests/test_sae_manifold_converged_latents_issue_357.py::test_converged_latents_exposes_t_star_and_a_star` →
```
thread '<unnamed>' panicked at crates/gam-sae/src/manifold/construction.rs:3812:53:
index out of bounds: the len is 3 but the index is 3
  3: gam_sae::manifold::construction::…::assemble_arrow_schur_inner
  4: gam_sae::manifold::fit_drivers::…::run_joint_fit_arrow_schur
  5: gam_sae::structure_harvest::run_production_structure_search
```

**Root cause.** The structure search grows K via Birth (`born_atom`) and Fission (`duplicate_atom`). Both push a new `log_ard` block onto the child `rho` but **never extend `log_lambda_smooth`** — a length-K per-atom vector (#1556) that the Arrow-Schur assembly indexes by `atom_idx` (`construction.rs`: `scaled_s[[i,j]] = lambda_smooth[atom_idx] * s_ij`). After a birth/fission lands, `atoms.len() == K+1` but `log_lambda_smooth.len() == K`, so the next assembly reads index K out of a len-K vec and panics.

This was masked because the only test exercising `apply_structure_move` (`apply_move_restructures_warm`) asserted `log_ard.len() == K+1` but **not** `log_lambda_smooth.len()`, and the assembly had a length guard for `log_ard` but not for `log_lambda_smooth`.

**Fix** (PR linked below): extend `log_lambda_smooth` in lockstep with `log_ard` in both `born_atom` and `duplicate_atom`; add a defensive length-K guard in `assemble_arrow_schur_inner`; new Rust regression drives the real assembly after a fission/birth (panics on pre-fix code); hardened `apply_move_restructures_warm` to pin the `log_lambda_smooth` length too.

**Comment by SauersML** [comment]
**Fleet agent status (from Discussions):** Agent `gam-closed-238-417` → PR #1739 (mergeable on main). The #357 topology-race REML fix (`log_lambda_smooth` grow-on-K; #977/#357) arrived on main via merged PR #1740, and the local duplicate (byte-equivalent) was dropped. PR #1739 still contributes on top of main:
1. The genuine #357 feature surface: `converged_latents()`/`project()`/`reconstruct()` + warm-start `a_init`/`t_init` (latent_basis_and_sae_ffi.rs).
2. A born-monomial-patch OOS round-trip fix: a structure-search BIRTH races a d=1 EuclideanPatch line at degree 3 (M=4); the OOS rebuild must recover that born degree from the trained decoder width (SAE_EUCLIDEAN_PATCH_RECOVERY_MAX_DEGREE=3) instead of freezing the seed degree 2, else Φ has 3 cols vs a 4-row decoder and reconstruct/project break on held-out matrices.
3. Centerless-patch `duchon_centers` synthesis for born EuclideanPatch/Linear/Poincaré atoms inheriting None centers from a periodic/sphere seed.
4. Regression tests (incl. driving the real arrow-Schur assembly on a grown dictionary).

Verified on the merged tree: gam-sae structure_harvest 19/19; Python issue-357 (5) + issue-1229 (2) = 7/7. **Flagged adjacent (out of scope):** over-birth — a clean K=3 3-circle mixture grows to K=8-9 periodic atoms (right topology, too many; #1026 territory); and a fit-time outer-ρ cost-stall `RemlConvergenceError` on n=80/seed=7 (same family as #1744).
